pkg/apis/core/v1/validation/validation.go

```diff
@@ -141,6 +148,13 @@ func ValidatePodLogOptions(opts *v1.PodLogOptions) field.ErrorList {
 		if *opts.SinceSeconds < 1 {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("sinceSeconds"), *opts.SinceSeconds, "must be greater than 0"))
 		}
+	case opts.Stream != nil:
+		if !supportedLogStreamType.Has(*opts.Stream) {
+			allErrs = append(allErrs, field.NotSupported(field.NewPath("stream"), *opts.Stream, supportedLogStreamType.UnsortedList()))
+		}
+		if *opts.Stream != v1.LogStreamTypeAll && opts.TailLines != nil {
```

Comment:
will `opts.Stream == nil` be an option or when not specified it will be defaulted to `all`?

Reply:
`opts.Stream` can be nil: https://github.com/knight42/kubernetes/blob/c78307c50b1e77f8f98161b1fc054be8f7f16a21/pkg/registry/core/pod/strategy.go#L496-L499

If `opts.Stream` is nil, it will be considered as `All`: https://github.com/knight42/kubernetes/blob/c78307c50b1e77f8f98161b1fc054be8f7f16a21/pkg/kubelet/server/server.go#L739-L744

Reply:
yes, but my question is whether anything will default it to All before this call?

Reply:
> whether anything will default it to All before this call?

I believe nothing will default it to All before validation, unless the param decoder behaves unexpectedly.

Reply:
oh, i see. This entire block only executes in `case opts.Stream != nil:` This makes sense to me now. Thank you for confirming